src/dev/flang/fuir/analysis/dfa/DFA.java

```diff
@@ -1660,35 +1660,53 @@ void hot(Call c)
    */
   void analyze(Call c)
   {
-    if (_fuir.clazzKind(c.calledClazz()) == FUIR.FeatureKind.Routine)
+    int cc = c.calledClazz();
+    if (_fuir.clazzKind(cc) == FUIR.FeatureKind.Routine)
       {
         var i = c._instance;
         check
-          (c._args.size() == _fuir.clazzArgCount(c.calledClazz()));
+          (c._args.size() == _fuir.clazzArgCount(cc));
         for (var a = 0; a < c._args.size(); a++)
           {
-            var af = _fuir.clazzArg(c.calledClazz(), a);
+            var af = _fuir.clazzArg(cc, a);
             var aa = c._args.get(a);
             i.setField(this, af, aa.value());
           }
 
         // copy outer ref argument to outer ref field:
-        var or = _fuir.clazzOuterRef(c.calledClazz());
+        var or = _fuir.clazzOuterRef(cc);
         if (or != NO_CLAZZ)
           {
             i.setField(this, or, c.target());
           }
 
         var ai = new AbstractInterpreter2<Val>(_fuir, new Analyze(c));
-        var r = ai.processClazz(c.calledClazz());
+        var r = ai.processClazz(cc);
         if (r != null)
           {
-            c.returns();
+            var rf = _fuir.clazzResultField(cc);
+            c.returns(hasNonUnitResultField(cc, rf)
+              ? c._instance.readField(this, rf, NO_SITE, c)
+              : r);
```

Comment:
I think it would be nice if this code would not duplicate part of the code from Call.java:336ff but would set `_result` to `_instance` in case we call a constructor, i.e., move the code from Call.java to here (or to a separate feature `Call.getResult` or similar.